src/main/scala/lms/transformation/distributed_tensor/distributedTensorUnary.scala

```diff
@@ -45,6 +63,30 @@ trait FixedSizeDistributedTensorUnaryTypeLess extends FixedSizeDistributedTensor
         (() => {
           Accumulate(gradMap(a), Negate(gradMap(s), anno), anno); ()
         }) +=: backwardNodes
+     
+    case Node(s, "tensor_invert", tt::Backend.Const(anno:Anno)::(a:Backend.Sym)::_, _) =>
+        implicit val pos = Adapter.oldSourceMap(s)
+        forwardNodes += node
+
+        (() => {
+          Accumulate(gradMap(a), Invert(gradMap(s), anno), anno); ()
+        }) +=: backwardNodes
+    
+    case Node(s, "tensor_tanh", tt::Backend.Const(anno:Anno)::(a:Backend.Sym)::_, _) =>
+        implicit val pos = Adapter.oldSourceMap(s)
+        forwardNodes += node
+
+        (() => {
+          Accumulate(gradMap(a), Tanh(gradMap(s), anno), anno); ()
```

Comment:
the derivative of tanh is a little more complex, but you can refer to this https://blogs.cuit.columbia.edu/zp2130/derivative_of_tanh_function/

Actually the computations here are fairly complex, so that maybe we can define a TanhGrad node that just do that. TanhGrad is likely a binary op.

Reply:
I looked at the tanh derivative. I get why it needs to be different, but why would it need to be a binary op? The derivate still only takes 1 variable

Reply:
I updated how the tanh gradient is calculated. I didn't make its own nodes because I didn't understand the value in doing so. I can't imagine how we would ever reuse the tanh derivative.